src/index.ts

```diff
@@ -109,27 +109,35 @@ class PlanetScaleDriver implements Driver {
 const sharedConnections = new WeakMap<PlanetScaleDialectConfig, Connection>()
 
 class PlanetScaleConnection implements DatabaseConnection {
-  #config: PlanetScaleDialectConfig
-  #conn: Connection
-  #transactionClient?: PlanetScaleConnection
+  #client: Client
+  #transactionConn?: Connection
+  #useSharedConnection: boolean
 
-  constructor(config: PlanetScaleDialectConfig, isForTransaction = false) {
-    this.#config = config
-    const useSharedConnection = config.useSharedConnection && !isForTransaction
-    const sharedConnection = useSharedConnection ? sharedConnections.get(config) : undefined
-    this.#conn = sharedConnection ?? connect({cast: inflateDates, ...config})
-    if (useSharedConnection) sharedConnections.set(config, this.#conn)
+  get #config(): Config {
+    return this.#client.config
+  }
+
+  constructor(client: Client, useSharedConnection = false, isForTransaction = false) {
+    this.#client = client
+    this.#useSharedConnection = useSharedConnection && !isForTransaction
+    if (this.#useSharedConnection) sharedConnections.set(this.#config, this.#client.connection())
```

Comment:
Small change here, otherwise the shared connection will get reset whenever a new `PlanetScaleConnection` is constructed:

```suggestion
    if (this.#useSharedConnection) sharedConnections.set(this.#config, sharedConnections.get(this.#config) ?? this.#client.connection())
```

Reply:
ah good catch.